From cacf637c8e503b2096881360010de057e334fe69 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 7 Apr 2014 00:04:38 -0700
Subject: [PATCH] Three fixes for permissions

- Add debug printing when allowing someone
- Return True or False from has_permission rather than the builtin 'input'
- Add a few more methods to get stuff

diff --git a/core/permissions.py b/core/permissions.py
index 103542a..25103d1 100644
--- a/core/permissions.py
+++ b/core/permissions.py
@@ -44,5 +44,21 @@ class PermissionManager(object):
 
         for pattern in allowed_users:
             if fnmatch(mask.lower(), pattern.lower()):
-                return input
+                self.logger.info("Allowed user {} access to {}".format(mask, perm))
+                return True
 
+        return False
+
+    def get_group_permissions(self, group):
+        return self.group_perms.get(group)
+
+    def get_group_users(self, group):
+        return self.group_users.get(group)
+
+    def get_user_permissions(self, user):
+        permissions = []
+        for permission, users in self.perm_users.items():
+            for mask in users:
+                if fnmatch(user.lower(), mask.lower()):
+                    permissions.append(permission)
+        return permissions
-- 
1.9.1

